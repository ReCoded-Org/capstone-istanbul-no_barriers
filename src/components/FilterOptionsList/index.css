:root {
    --main-filter-option-bg-color: #E7E7E7;
    --main-filter-option-text-color: #00374d;
    --orange-link: #f2c94c;
  }

  .filterOptionsList {
      background-color: var(--main-filter-option-bg-color);
      color: var(--main-filter-option-text-color);
  }

  .filterOption:not(:last-of-type) {
      border-bottom: 1px solid var(--main-filter-option-text-color);
  }

  .filterOptionPlusSign {
      margin-left: 1rem;
  }