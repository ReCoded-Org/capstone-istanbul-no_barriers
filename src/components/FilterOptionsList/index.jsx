import React from "react";
import "./index.css";

// This renders the filter options of a given filter as parameter.
// It is called in AdvancedFilters component
const FilterOptionsList = ({ filter, setChosenOptions, filterCategory }) => {
  return (
    <div className="filterOptionsList">
      {filter.map((option) => {
        return (
          <div
            className="filterOption d-flex align-items-center p-1"
            key={option}
            onClick={() =>
              setChosenOptions((state) => {
                state[filterCategory].push(option);
              })
            }
          >
            <span className="filterOptionPlusSign">+</span>
            <p className="filterOptionTitle ">{option}</p>
          </div>
        );
      })}
    </div>
  );
};

export default FilterOptionsList;
