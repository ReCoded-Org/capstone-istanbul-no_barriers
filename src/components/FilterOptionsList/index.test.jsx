import React from "react";
import renderer from "react-test-renderer";
import FilterOptionsList from "./index";

const options = { services: ["education"], cities: ["Istanbul", "Ankara"] };
const optionsKeys = Object.keys(options);
it("FilterOptionsList renders correctly", () => {
  const tree = renderer
    .create(
      optionsKeys.map((filterKey) => (
        <FilterOptionsList filter={options[filterKey]} key={filterKey} />
      ))
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
