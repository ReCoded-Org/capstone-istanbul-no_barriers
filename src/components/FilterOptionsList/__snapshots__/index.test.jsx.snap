// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterOptionsList renders correctly 1`] = `
Array [
  <div
    className="filterOptionsList"
  >
    <div
      className="filterOption d-flex align-items-center p-1"
      onClick={[Function]}
    >
      <span
        className="filterOptionPlusSign"
      >
        +
      </span>
      <p
        className="filterOptionTitle "
      >
        education
      </p>
    </div>
  </div>,
  <div
    className="filterOptionsList"
  >
    <div
      className="filterOption d-flex align-items-center p-1"
      onClick={[Function]}
    >
      <span
        className="filterOptionPlusSign"
      >
        +
      </span>
      <p
        className="filterOptionTitle "
      >
        Istanbul
      </p>
    </div>
    <div
      className="filterOption d-flex align-items-center p-1"
      onClick={[Function]}
    >
      <span
        className="filterOptionPlusSign"
      >
        +
      </span>
      <p
        className="filterOptionTitle "
      >
        Ankara
      </p>
    </div>
  </div>,
]
`;
